### PR TITLE
Bug 1690592: Update API group and version

### PR DESCRIPTION
**Pull request by aravindhp**
- [csc] Add missing deepcopy code
- Update API group and version
- Move to using new operator-registry command

### CODE REVIEW

**Comment by aravindhp** [comment]
This is early preview of the API group and version changes. There will be an additional commit once @tkashem opens a PR against `operator-registry` to move away from using `OperatorSources` directly.

File: deploy/upstream/03_operatorsource.crd.yaml [hunk 1]
```diff
@@ -29,6 +29,14 @@ spec:
...
     description: App registry namespace
     JSONPath: .spec.registryNamespace
+  - name: DisplayName
+    type: string
+    description: Display (pretty) name to indicate the OperatorSource's name
+    JSONPath: .spec.displayName
+  - name: Publisher
+    type: string
+    description: Publisher of the OperatorSource
+    JSONPath: .spec.publisher
```
Comment (kevinrizza) [inline deploy/upstream/03_operatorsource.crd.yaml]: Why are we adding these in this pr?

Comment (aravindhp) [reply deploy/upstream/03_operatorsource.crd.yaml]: I copy pasted from the manifests folder file and this showed up as a discrepancy. I am fine removing this and fixing it in a separate PR. LMK what you and the other reviewers prefer.

Comment (kevinrizza) [reply deploy/upstream/03_operatorsource.crd.yaml]: No strong preference here, if it matches the manifests folder I'm fine with leaving it

**Comment by alecmerdler** [comment]
@aravindhp The way we are handling this for OLM is through a 2-phase migration:
1. Update OLM Operator to support both APIs
2. Update `openshift/console` to consume new API
3. Remove old API from OLM Operator

Can you update this PR to support both APIs so that the console continues to work?

**Comment by aravindhp** [comment]
>Can you update this PR to support both APIs so that the console continues to work?

@alecmerdler I am in the process of doing that. Once I do that will you handle the console changes? Or do you need @galletti94 to help out?

**Comment by alecmerdler** [comment]
@aravindhp I have the console changes ready in a branch.

**Comment by aravindhp** [comment]
@tkashem @kevinrizza @awgreene @anik120 please review

**Comment by kevinrizza** [comment]
/lgtm

@aravindhp thinking out loud here, do we have any example CRs out in the wild that need updating? It seems like everything in our source is updated here, but is there anything anywhere outside that we need to make sure is updated? I'm assuming at least OpenShift docs will need to be updated. Is there anything in community-operators that refers to creating `operatorsources`?

**Comment by aravindhp** [comment]
@kevinrizza we will have to do a sweep and update it where needed.

@chrisnegus if this PR does get merged for beta 3 then the docs have to be updated. We changed our `Group` and `Version` to `operators.coreos.com` and `v1`.

**Comment by chrisnegus** [comment]
@aravindhp Okay. Let me know if the change happens and I'll update the docs.

**Comment by kevinrizza** [comment]
@aravindhp can you actually update your comment to mention the change to update the operator-registry command?

**Comment by aravindhp** [comment]
> @aravindhp can you actually update your comment to mention the change to update the operator-registry command?

Done

File: pkg/catalogsourceconfig/registry.go [hunk 2]
```diff
@@ -235,26 +243,28 @@ func (r *registry) getLabel() map[string]string {
...
 		if err != nil {
 			r.log.Errorf("Error %v reading package %s", err, packageID)
 			continue
 		}
-		opsrcNamespacedName := opsrcMeta.Namespace + "/" + opsrcMeta.Name
-		if !strings.Contains(opsrcString, opsrcNamespacedName) {
-			opsrcString += opsrcNamespacedName + ","
-			opsrcList = append(opsrcList, opsrcMeta.Name)
+		//--registry="https://quay.io/cnr|community-operators" --regisry="https://quay.io/cnr|custom-operators|mynamespace/mysecret"
+		appRegistry := "--registry=" + opsrcMeta.Endpoint + "|" + opsrcMeta.RegistryNamespace
```
Comment (tkashem) [inline pkg/catalogsourceconfig/registry.go]: looks like the pipe delimited soure is not enclosed in double quotes `--registry="https://quay.io/cnr|community-operators"`

**Comment by aravindhp** [comment]
> @aravindhp Okay. Let me know if the change happens and I'll update the docs.

@chrisnegus this PR has merged. Please update the docs to reflect the new group version `operators.coreos.com/v1`.

**Pull request merged**